device.go

```diff
@@ -0,0 +1,152 @@
+//
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/kata-containers/agent/pkg/uevent"
+	pb "github.com/kata-containers/agent/protocols/grpc"
+	"github.com/sirupsen/logrus"
+)
+
+type deviceDriversHandler func(device pb.Device, spec *pb.Spec) error
+
+var deviceDriversHandlerList = map[string]deviceDriversHandler{
+	"blk": blockDeviceHandler,
+}
+
+func blockDeviceHandler(device pb.Device, spec *pb.Spec) error {
```

Comment:
Now this is no longer block device specific, maybe just a call it `genericDeviceHandler`?

Reply:
Not sure I get your point here. We still need a `deviceHandler` for each type of device that is going to be passed, right ?

Reply:
True, but `blockDeviceHandler` is not really block device specific now, right? I was hoping that we can use it to handle some other types of devices as well. There are two ways of doing it:
1. rename `blk` to a more generic name for devices
2. define other types of `deviceDriversHandler` and use the same handler to handle them

I'm OK with either way but the handler function name needs also to be generic.

Anyway it is just a minor issue. We can get it in as is and change it later when adding other device types support.